src/main/java/frc/robot/commands/PositionRecorder.java

```diff
@@ -0,0 +1,119 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import edu.wpi.first.networktables.EntryListenerFlags;
+import edu.wpi.first.networktables.EntryNotification;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.DriveTrain;
+
+public class PositionRecorder extends CommandBase {
+  /**
+   * Creates a new PositionRecorder.
+   */
+  public boolean isRunning = false;
+
+  DriveTrain drivetrain;
+  PrintWriter writer;
+  long start;
+
+  String directoryName="position-recordings";
+
+  NetworkTableEntry directoryNameEntry;
+  NetworkTableEntry isRunningEntry;
+  
+  public PositionRecorder(DriveTrain drivetrain) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    initSmartDashboard();
+    this.drivetrain = drivetrain;
+  }
+  public void initSmartDashboard(){
+    ShuffleboardTab tab =Shuffleboard.getTab("Position Recorder");
+    isRunningEntry=tab.add("Is running", isRunning).withWidget("Toggle Button").getEntry();
+    directoryNameEntry=tab.add("Directory Name", directoryName).getEntry();
+
```

Comment:
I don't think this make sense. What if I am not running Shuffleboard? Why not just use standard SmartDashboard entries?

Reply:
I think we can leave this for now.